crates/cli-support/src/js/binding.rs

```diff
@@ -1450,8 +1470,11 @@ fn adapter2ts(ty: &AdapterType, dst: &mut String, refs: Option<&mut HashSet<TsRe
         AdapterType::Bool => dst.push_str("boolean"),
         AdapterType::Vector(kind) => dst.push_str(&kind.js_ty()),
         AdapterType::Option(ty) => {
-            adapter2ts(ty, dst, refs);
-            dst.push_str(" | undefined");
+            adapter2ts(ty, position, dst, refs);
+            dst.push_str(match position {
+                TypePosition::Argument => " | undefined | null",
```

Comment:
```suggestion
                TypePosition::Argument => unreachable!(),
```
We always handle this case manually, right?

Reply:
We don't for variadic arguments (`bindings.rs` line 419). Though I'm not sure how relevant that is, since varargs probably don't allow `Option<T>`.

But I also don't like the idea of `adapter2ts` knowing with what types it is called. This feels like a bit of global reasoning to me.

Reply:
Hm, aren't we handling all `Option<T>` manually before they ever arrive in `adapter2ts()`?
I'm just concerned because this code is actually incorrect, this function can't handle this case correctly.

Ideally we move `Option<T>` handling entirely into this function, but this could be done in a separate PR.

So unless I'm missing something, I think the whole `AdapterType::Option(ty) => { ... }` statement should yield `unreachable!()` imo.

Reply:
> I'm just concerned because this code is actually incorrect, this function can't handle this case correctly.

Lets say varargs do support this, would the output be correct?

Reply:
Yes, the output would be correct for [documenting varargs in JSDoc](https://stackoverflow.com/questions/4729516/correct-way-to-document-open-ended-argument-functions-in-jsdoc) AFAICT. E.g. `Option<u32>` would produce:

```
@param {...number | null | undefined} name
```

Reply:
After thinking about it some more, let's go with your suggestion. Varargs are pretty under-tested and under-documented right now, so I have no idea whether they even allow `Option<T>`. 

I will need to re-add this in the future (because I plan to add support for arrays soon), but I can re-add it when there are tests for it then.